lib/pliny/librato/metrics/backend.rb

```diff
@@ -31,14 +35,15 @@ def start
 
         def stop
           metrics_queue.push(POISON_PILL)
-          timer.terminate
+          # Ensure timer is not running when we terminate it
+          sync { timer.terminate }
```

Comment:
I'll add a note in the README that this won't work with `Signal#trap` as you can't obtain a mutex in a trap.